src/smallmap_gui.cpp

```diff
@@ -829,7 +834,7 @@ class SmallMapWindow : public Window {
 
 		if (map_type == SMT_LINKSTATS) this->overlay->SetDirty();
 		if (map_type != SMT_INDUSTRY) this->BreakIndustryChainLink();
-		this->SetDirty();
+		this->ReInit();
```

Comment:
@2TallTyler Thanks for the review and the pointer to `build_vehicle_gui.cpp`!

Glad to know the `NWID_SELECTION` and `SZSP_NONE` combo is the standard ways to show/hide things.

I initially tried that route after looking at your PR for showing/hiding block signals #11897. But ran into a weird rendering issue where the now-void space isn't resized properly:

![screenshot showing an empty space rendering incorrectly](https://github.com/OpenTTD/OpenTTD/assets/1967998/6af7095b-7697-497d-9ac1-8f87d1a251ae)

So I dug around other call sites and eventually found that I need to use `ReInit()` when using `StackedZeroSizePlanes`:

https://github.com/OpenTTD/OpenTTD/blob/e802c1a97fac5bfad5ff460ea8dfd37d24571602/src/widget_type.h#L495-L498

Let me know if `ReInit` should be avoided here, or if there's a more scoped method like `ResizeWindow` (tho that's private).

Reply:
If `ReInit` is used elsewhere, it's good enough for here! I don't imagine it will be clicked very often, so it's cheap. 🙂